qa/tasks/mgr/dashboard/__init__.py

```diff
@@ -0,0 +1 @@
+DEFAULT_VERSION = '1.0'
```

Comment:
As both files (__init__.py) are in different paths (/qa/tasks/dashboard & /dashboard) , I declared the variable separately

Reply:
> What about putting this into the `dashboard/controller/__init__.py` or `dashboard/__init__.py` instead of in the QA code?
> 
> Additionally, given the amount of parsing-printing-formatting required around REST API version, why not coming with a high level class for handling versioning (parsing, etc):
> 
Sounds great!
> ```python
> from collections import namedtuple
> import re
> 
> class ApiVersion(namedtuple('ApiVersion', ['major', 'minor'], defaults=(1,0))):                      
>     MIME_TYPE = 'application/vnd.ceph.api.v{}.{}+json'                                               
>     RE_PATTERN = re.compile('application/vnd\.ceph\.api\.v(\d+)(?:\.(\d))?\+json')                        
>                                                                                                      
>     def __str__(self):                                                                               
>         return self.MIME_TYPE.format(self.major, self.minor)                                         
>                                                                                                      
>     def to_mime(self):                                                                               
>         return str(self)                                                                             
>                                                                                                      
>     @classmethod                                                                                     
>     def from_mime(cls, mime):                                                                             
>         try:
>             return cls._make(                                                                            
>                 map(int, cls.RE_PATTERN.match(mime).groups('0'))                                          
>             )
>         except AttributeError:
>             return None
> 
> >>> ApiVersion()
> ApiVersion(major=1, minor=0)
> 
> >>> str(ApiVersion())
> 'application/vnd.ceph.api.v1.0+json'
> 
> >>> ApiVersion(1,2).to_mime()
> 'application/vnd.ceph.api.v1.2+json'
> 
> >>> ApiVersion.from_mime('application/vnd.ceph.api.v1.2+json')
> ApiVersion(major=1, minor=2)
> 
> >>> ApiVersion.from_mime('application/vnd.ceph.api.v1+json')
> ApiVersion(major=1, minor=0)
> 
> >>> ApiVersion.from_mime('application/vnd.ceph.api.v1.+json')
> None
> 
> >>> ApiVersion(1,0) < ApiVersion(1,1)
> True
This looks great to me. It looks like a generalized approach to me. Should we take this to another PR(naming this PR as default versioning) or maybe just under this PR and a separate commit?

Reply:
That snippet is ready to work, so if you want, you could just try and replace a few occurrences, and if this simplifies the code and works as expected (that's the good part of having unit tests) we could go for this approach on the rest of the PR. What do you think about it?

Reply:
Yes, sure! Will try to apply this approach to unit tests

Reply:
As @epuertat has agreed to take this approach in a later PR , I'm resolving this conversation. The later PR will include a seperate model or helper file in order to make it work for qa tasks and dashboard modules/unit-tests too. And for follow up will create a tracker for this too.

Reply:
Created the tracker for the later improvements https://tracker.ceph.com/issues/48015